src/SFML/Graphics/ConvexShape.cpp

```diff
@@ -66,4 +66,44 @@ Vector2f ConvexShape::getPoint(std::size_t index) const
     return m_points[index];
 }
 
+////////////////////////////////////////////////////////////
+Vector2f ConvexShape::getGeometricCenter() const
+{
+    if (m_points.size() > 2)
+    {
+        const auto& [offx, offy] = m_points.front();
+        float x = 0.f, y = 0.f;
+        float twicearea = 0.f;
+
+        float f;
+
+        for (auto it = m_points.begin(), prev = std::prev(m_points.end()); it != m_points.end(); prev = it++)
+        {
+            const auto& [p1x, p1y] = *it;
+            const auto& [p2x, p2y] = *prev;
+
+            f = (p1x - offx) * (p2y - offy) - (p2x - offx) * (p1y - offy);
+            twicearea += f;
+            x += (p1x + p2x - 2.f * offx) * f;
+            y += (p1y + p2y - 2.f * offy) * f;
+        }
+
+        f = twicearea * 3.f;
+
+        return Vector2f(x / f + offx, y / f + offy);
+    }
+
+    if (m_points.size() == 2)
+    { // a line
+        const auto& [x1, y1] = m_points.front();
+        const auto& [x2, y2] = m_points.back();
+        return Vector2f((x1 + x2) / 2.f, (y1 + y2) / 2.f);
+    }
+
+    if (not m_points.empty())
+        return m_points.front(); // a single point
+
+    return Vector2f{}; // empty
```

Comment:
Not too sure about this one. It's impossible to differentiate between a polygon whose geometrical center is in `0,0` and a polygon with no points. Maybe this function should return `std::optional<Vector2f>`?

Reply:
That's an option or it could throw an exception or leave the case UB

Reply:
@Bromeon  Sounds like a good idea. What to do in `NDEBUG` builds? `std::terminate`?